test/extensions/tracers/xray/util_test.cc

```diff
@@ -0,0 +1,66 @@
+#include "extensions/tracers/xray/util.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace XRay {
+
+TEST(XRayWildcardTest, matchingEmpty) {
```

Comment:
Thanks; this is looking easier to read. Per earlier comment, WDYT about adding some/all of the testcases from https://github.com/apache/incubator-pagespeed-mod/blob/master/pagespeed/kernel/base/wildcard_test.cc
?

Reply:
I did. Even the CI is breaking on the "mispell" that I copied from there.
Let me know if there are specific test cases that you think I missed. I didn't copy everything because many were overlapping with what was there. But happy to add any that you think are important or I missed.

Reply:
oh cool. how about some of  the cases in `TEST_F(WildcardTest, LengthAtLeastTwo)` ?

Reply:
I'll add the following test cases:
```cpp
  EXPECT_FALSE(WildcardMatch("??*", "a"));
  EXPECT_TRUE(WildcardMatch("??*", "aa"));
  EXPECT_TRUE(WildcardMatch("??*", "aaa"));
```